aperturedb/cli/adb.py

```diff
@@ -0,0 +1,23 @@
+import typer
+
+from aperturedb.cli import configure, ingest, utilities, transact
+
+app = typer.Typer()
+
+app.add_typer(ingest.app, name="ingest", help="Ingest data into ApertureDB.")
+app.add_typer(configure.app, name="config",
+              help="Configure ApertureDB client.")
+app.add_typer(utilities.app, name="utils", help="Utilities")
+app.add_typer(transact.app, name="transact",
+              help="Run a transaction against ApertureDB.")
+
+
+@app.callback()
+def check_context(ctx: typer.Context):
+    if ctx.invoked_subcommand != "config":
+        configure.check_configured(as_global=False) or \
+            configure.check_configured(as_global=True, show_error=True)
+
```

Comment:
I looked at the documentation a bit for Typer, and they seem to mainly use this for injecting non-command specific options. They name this `callback` but it seems like  it is a hook that runs around parsing time.

What seems confusing is that they use it to add commands for the main program, which would make it pre-parse but you seem to be using it after that has happened, which would make it post-parse.

Is it both and depends on the args you want it to send?

Reply:
ok, so I think I've answered my question; the main entry is https://typer.tiangolo.com/tutorial/commands/callback/ but then link tohttps://typer.tiangolo.com/tutorial/commands/context/ which mentions that callback will only be called right before a command by default.

Which is confusing because the first link shows it creating a command line argument --verbose, but I assume the decorator makes it process that.

So now it all makes sense.